File: project-template/spring-boot-library-monorepo-java-gradle/spring-lib-sample/spring-lib-test-sample/src/main/java/acktsap/SomeComponent.java
package acktsap;

import javax.annotation.PostConstruct;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import acktsap.spring.lib.Library;

@Component
class SomeComponent {
    @Autowired
    private Library library;

    @PostConstruct
    void init() {
        System.out.println("library.someLibraryModuleMethod: " + library.someLibraryModuleMethod());
    }
}

File: project-template/spring-boot-library-monorepo-gradle/spring-lib-sample/spring-lib-kotlin-sample/src/main/kotlin/acktsap/SomeComponent.kt
package acktsap

import acktsap.spring.lib.Library
import org.springframework.stereotype.Component
import javax.annotation.PostConstruct

@Component
open class SomeComponent(
    private val library: Library
) {
    @PostConstruct
    fun init() {
        println("library.someExtension: ${library.someExtension()}")
    }
}
